shared/reports/resources.py

```diff
@@ -598,7 +593,17 @@ def build_files(files: dict[str, Any]) -> dict[str, ReportFileSummary]:
     # NOTE: this mutates `files` in-place
     for filename, file_summary in files.items():
         if not isinstance(file_summary, ReportFileSummary):
-            files[filename] = ReportFileSummary(*file_summary)
+            chunks_index = file_summary[0]
+            file_totals = file_summary[1]
+
+            if len(file_summary) > 2:
+                # Old reports have `session_totals` but new reports omit it
+                # Graph `diff_totals` from the back of the list either way
+                diff_totals = file_summary[-1]
+            else:
+                diff_totals = None
```

Comment:
Just to double-check:
The format has always been serialized as a 4-element tuple? So there is no backwards compatibility with very ancient reports that have a 3-element `chunks_index, file_totals, session_totals` floating around?

If such backwards compatibility is a concern, I would suggest to keep this as 4-element tuple, but just use an empty value for the `session_totals`.

Reply:
https://github.com/codecov/codecov.io/commit/d0c28c0899a732a47db9882aa73cbb8d580161ce this appears to be the commit where the field was introduced. i did not see any usages of it lol (the `sum_session_totals` SQL function in that same commit works on the `"sessions"` key in the report JSON, not these session totals. and also the sql function is deleted now by a later migration)

you are right about the backcompat, i will update